File: docs/recognition/static/recognition/js/services/overlay-renderer.js
class OverlayRenderer {
    constructor(ctx) {
        this.ctx = ctx;
    }

    render_detections(faces, width, height) {
        this.ctx.clearRect(0, 0, width, height);
        
        faces.forEach(face => {
            const isRecognized = face.name !== 'Unknown';
            
            // Set colors based on recognition status
            const strokeColor = isRecognized ? '#00ff00' : '#ff0000';
            const bgColor = isRecognized ? 'rgba(0, 255, 0, 0.2)' : 'rgba(255, 0, 0, 0.2)';
            const textColor = isRecognized ? '#00ff00' : '#ff0000';
            
            // Draw face rectangle with background
            this.ctx.fillStyle = bgColor;
            this.ctx.fillRect(face.x, face.y, face.width, face.height);
            
            this.ctx.strokeStyle = strokeColor;
            this.ctx.lineWidth = 2;
            this.ctx.strokeRect(face.x, face.y, face.width, face.height);
            
            // Draw name label
            const label = face.confidence ? `${face.name} (${face.confidence}%)` : face.name;
            this.ctx.font = '16px Arial';
            const textWidth = this.ctx.measureText(label).width;
            
            // Label background
            this.ctx.fillStyle = 'rgba(0, 0, 0, 0.7)';
            this.ctx.fillRect(face.x, face.y - 25, textWidth + 10, 25);
            
            // Label text
            this.ctx.fillStyle = textColor;
            this.ctx.fillText(label, face.x + 5, face.y - 7);
        });
    }

    clear() {
        if (this.ctx) {
            this.ctx.clearRect(0, 0, this.ctx.canvas.width, this.ctx.canvas.height);
        }
    }
}

File: recognition/utils/video/overlay_renderer.py
class OverlayRenderer:
    def __init__(self, canvas_context):
        self.ctx = canvas_context
        
    def render_detections(self, faces, canvas_width, canvas_height):
        """Render face detection results on canvas overlay"""
        self.ctx.clearRect(0, 0, canvas_width, canvas_height)
        
        for face in faces:
            self._render_face(face)
            
    def _render_face(self, face):
        """Render a single face detection"""
        is_recognized = face['name'] != 'Unknown'
        
        # Set colors based on recognition status
        stroke_color = '#00ff00' if is_recognized else '#ff0000'
        bg_color = 'rgba(0, 255, 0, 0.2)' if is_recognized else 'rgba(255, 0, 0, 0.2)'
        text_color = '#00ff00' if is_recognized else '#ff0000'
        
        # Draw face rectangle with background
        self.ctx.fillStyle = bg_color
        self.ctx.fillRect(face['x'], face['y'], face['width'], face['height'])
        
        self.ctx.strokeStyle = stroke_color
        self.ctx.lineWidth = 2
        self.ctx.strokeRect(face['x'], face['y'], face['width'], face['height'])
        
        # Draw name label
        label = f"{face['name']} ({face['confidence']}%)" if face['confidence'] else face['name']
        self.ctx.font = '16px Arial'
        text_width = self.ctx.measureText(label).width
        
        # Label background
        self.ctx.fillStyle = 'rgba(0, 0, 0, 0.7)'
        self.ctx.fillRect(face['x'], face['y'] - 25, text_width + 10, 25)
        
        # Label text
        self.ctx.fillStyle = text_color
        self.ctx.fillText(label, face['x'] + 5, face['y'] - 7)